internal/api/internal/server.go

```diff
@@ -0,0 +1,57 @@
+package internal
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/go-chi/chi/v5"
+	"github.com/hashicorp/consul-api-gateway/internal/api/middleware"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/go-hclog"
+)
+
+//go:generate oapi-codegen -config ../schemas/internal.config.yaml ../schemas/internal.yaml
+
+var _ ServerInterface = &Server{}
+
+type Server struct {
+	logger       hclog.Logger
+	consulClient *api.Client
+}
+
+// TODO(andrew): most of this is boilerplate that should be generated
+
+func NewServer(url string, consulClient *api.Client, logger hclog.Logger) http.Handler {
```

Comment:
There are three different functions named NewServer across these packages that seem to do slightly different variations of server/handler setup, and I'd appreciate some header comments clarifying the difference between the three, but otherwise seems solid to me.

Reply:
Yep, we can add some docstrings as we go. The main thing is that we're composing two APIs together. Each API, internal & v1 are structured as their own API "servers" in that you can easily run them as standalone services (as we do in the integration-level tests here). The top-level `api/server.go` implementation just composes them together and should pretty much be what we use as the main thing we invoke in our entrypoint.

Reply:
The clients, btw are pretty much structured the same way. Mind if I just follow-up with some better doc comments as I flesh out the scaffolding?